rofl-client/ts/src/roflClient.ts

```diff
@@ -0,0 +1,284 @@
+        if (!this.url) return ROFL_SOCKET_PATH;
+        return '';
+    }
+
+    private async appdRequest(
+        method: 'GET' | 'POST',
+        path: string,
+        payload?: unknown,
+    ): Promise<unknown> {
+        const headers: Record<string, string> = {
+            'User-Agent': this.userAgent,
+        };
+        let body: string | undefined;
+
+        if (method === 'POST') {
+            body = JSON.stringify(payload ?? {});
+            // Canonical header casing for broader compatibility.
+            headers['Content-Type'] = 'application/json';
+            headers['Content-Length'] = Buffer.byteLength(body).toString();
+        }
+
+        const transportRequest: TransportRequest = this.isHttpUrl()
+            ? {
+                  method,
+                  path,
+                  baseUrl: this.url,
+                  payload,
+                  body,
+                  timeoutMs: this.timeoutMs,
+                  headers,
+              }
+            : {
+                  method,
+                  path,
+                  socketPath: this.socketPathOrDefault(),
+                  payload,
+                  body,
+                  timeoutMs: this.timeoutMs,
+                  headers,
+              };
+
+        const transport = this.transport ?? nodeTransport;
+        const {status, body: responseBody} = await transport(transportRequest);
+
+        if (status >= 400) {
+            const message =
+                typeof responseBody === 'string' ? responseBody : JSON.stringify(responseBody);
+            throw new Error(`ROFL request failed (${status}): ${message}`);
+        }
+
+        return responseBody;
+    }
+}
+
+/** Default Node.js transport with UDS + HTTP(S) support. */
+const nodeTransport: Transport = (req) =>
+    new Promise<TransportResponse>((resolve, reject) => {
+        const isHttp = !!req.baseUrl;
+        const isHttps = isHttp && req.baseUrl!.startsWith('https://');
+        const lib = isHttp ? (isHttps ? https : http) : http;
+
+        const onError = (err: Error) => reject(err);
+
+        const writeBody = (request: http.ClientRequest) => {
+            if (req.method === 'POST') {
+                const payload = req.body ?? JSON.stringify(req.payload ?? {});
+                request.write(payload);
+            }
+        };
+
+        const attachTimeout = (request: http.ClientRequest) => {
+            request.setTimeout(req.timeoutMs, () => {
+                request.destroy(new Error('ROFL request timed out'));
+            });
+        };
+
+        const onResponse = (res: http.IncomingMessage) => {
+            const chunks: Buffer[] = [];
+
+            res.on('data', (chunk) => {
+                chunks.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk));
+            });
+
+            res.on('end', () => {
+                const raw = Buffer.concat(chunks);
+                const text = raw.toString('utf8');
```

Comment:
I'm not familiar enough with low level requests to review. If this works it works :shrug: